### PR TITLE
Refactor <Table> and <Pagination> components

### LINKED ISSUE #221: [Task] Verify Table component
### Verify the CFPB Design System (React) component against the CFPB Design System
- [x] Has component intro text copied from the DS page
- [x] Has source link to component's DS page (ex - Source: https://cfpb.github.io/design-system/components/banner-notification)
- [x] Each DS variant is implemented as a separate story
  - [ ] Story name should be sentence case (ex. List Link => List link)
  - [ ] Naming is consistent with the DS
  - [ ] Same component names, same placeholder text, etc.
- [x] Order of stories matches between DS/DSR
- [x] Component is built correctly
  - [ ] Visually compare DSR implementation to DS
  - [ ] Verify that DS classes (organisms, molecules, atoms) are used, as opposed to styles defined in DSR
- [ ] Manual visual review has been conducted and component has passed this review

#### For new components 
- [ ] All steps for existing components plus the following steps
- [ ] The new component has been added to the CFPB Design System OR
- [ ] The CFPB Design System maintainers have been alerted that there is a new component that must be added to the system
- [ ] Documentation has been written for the new component (this is not a blocker for moving a component to verified)

### Run accessibility checks 
- [x] Component is keyboard-friendly (navigable with tab, space, enter, arrow keys, etc.)
- [x] Component does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)
- [ ] Component is screen reader friendly ([screen reader testing demo](https://drive.google.com/file/d/189A9Wx_T1C49TES5ZtYXlIFbJbLP472p/view))
  - [ ] Is all the meaningful visual information and text of the component conveyed by the screen reader? (text, non-decorative image descriptions, etc.)
  - [ ] When interacting with the component using a screen reader, do you have all the information you need to use it? (selected vs unselected, button vs link, expanded vs collapsed, etc.)
  - [ ] Does the component have similar screen reader behavior as the sample components in WCAG, the CFPB design system, WebAIM, or similar accessibility examples?
- [ ] For reference: [CFPB manual web accessibility audit](https://cfpb.github.io/design-system/guidelines/accessibility-audit-tools#cfpb-manual-web-accessibility-audit-1) 

### Verify component unit tests
- [ ] Verify component unit tests are implemented and passing - 85% or greater (ex: `yarn vitest Button`)
  
### Conduct PR review
- [ ] The component has been reviewed and approved by UI, UX, and FEWD

### Verify component
- [ ] Move component to verified in Storybook

### PR DESCRIPTION
**Pull request by meissadia**
Closes #221 

Dear Reviewer, not exactly a *small* PR, happy to talk through changes!  ~ M

## Changes

- Remove `react-table`
- Remove advanced Table capabilities (filter, sort)
- Encapsulated Pagination logic into a reusable hook: `usePagination`
- Sync Stories with Design System docs
- [Test] Table: Shows pagination when isPaginated
- [Test] usePagination: added unit tests

## How to test this PR

1. yarn vitest Table
2. yarn vitest Pagination

### CODE REVIEW

File: src/types/WidthPercent.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers */
+
+/**
+ * Values pulled from cfpb-core width utilities
+ * https://github.com/cfpb/design-system/blob/main/packages/cfpb-core/src/utilities.less#L328-L386
+ */
+export type WidthPercent =
```
Comment (billhimmelsbach) [inline src/types/WidthPercent.ts:7]: It's kind of a bummer we can't import these out of the design system. 🤔 I think we could refactor some DS stuff to make that work... but probably for another day. 👍

**Review by billhimmelsbach — COMMENTED** [review]
Heyo! This is looking great! I need a little more time to look over this, but I'm putting it on my calendar for this afternoon! 👍

**Review by billhimmelsbach — APPROVED** [review]
This is a great PR @meissadia! I added making the width percentages not hard coded to a [little post-launch issue](https://github.com/cfpb/design-system-react/issues/231), but otherwise I think it's looking good!

**Comment by meissadia** [comment]
@billhimmelsbach @shindigira A couple fixes for items I had missed:
- Sync Pagination header info with DS - [preview link](https://deploy-preview-225--cfpb-design-system-react.netlify.app/?path=/docs/components-draft-pagination--overview)
  Screenshot 2023-11-08 at 5 35 34 PM
- Add documentation for DSR's `usePagination` hook - [preview link](https://deploy-preview-225--cfpb-design-system-react.netlify.app/?path=/docs/components-draft-pagination--overview#usepagination)
  Screenshot 2023-11-08 at 5 35 23 PM
- Reduce the number of rows shown per page in the `Table - Pagination` story (from 20 to 5) to make it easier to interact with - [preview link](https://deploy-preview-225--cfpb-design-system-react.netlify.app/?path=/docs/components-draft-tables--overview#pagination)
  Screenshot 2023-11-08 at 5 35 08 PM

**Pull request merged**